From: Julian Ospald <hasufell@gentoo.org>
Date: Sun Mar 31 14:48:15 UTC 2013
Subject: build system

* not having gtk installed will lead to undefined macro
* intl subdir not needed, remove references
* po/Makefile.in will be generated by gettextize
* use modern icon destination
* don't install icons when gtk is disabled
* disable gconf when gtk is disabled
* install plugin header
* allow to pass specific python version

--- hexchat-2.9.4/configure.ac
+++ hexchat-2.9.4/configure.ac
@@ -115,8 +115,8 @@
         xlib=$enableval, xlib=yes)
 
 AC_ARG_ENABLE(python,
-	[AS_HELP_STRING([--disable-python],[don't build the python plugin])],
-        python=$enableval, python=yes)
+	[AS_HELP_STRING([--enable-python=pythonversion],[build the python plugin (default on, python2)])],
+        python=$enableval, python=python2)
 
 AC_ARG_ENABLE(perl,
 	[AS_HELP_STRING([--disable-perl],[don't build the perl plugin])],
@@ -200,13 +200,16 @@
 dnl ** GTK **************************************************************
 dnl *********************************************************************
 
-AM_PATH_GTK_2_0(2.10.0, havegtk=yes, havegtk=no)
-
-if test "$havegtk" = no; then
-	gtkfe=no
-	echo
-	echo Cannot find GTK\! Not building GTK FrontEnd.
-	echo
+# we might get undefined macro without this test
+if test "$gtkfe" = yes ; then
+	AM_PATH_GTK_2_0(2.10.0, havegtk=yes, havegtk=no)
+
+	if test "$havegtk" = no; then
+		gtkfe=no
+		echo
+		echo Cannot find GTK\! Not building GTK FrontEnd.
+		echo
+	fi
 fi
 
 if test "$gtkfe" != yes; then
@@ -222,7 +225,6 @@
 GUI_LIBS="$GUI_LIBS $GTK_LIBS"
 GUI_CFLAGS="$GUI_CFLAGS $GTK_CFLAGS"
 
-gnome=no
 #if test "$gnome" = yes; then
 #	AC_PATH_PROG(pkgconfigpath, pkg-config)
 #	AC_MSG_CHECKING(Gnome2 compile flags)
@@ -240,7 +242,11 @@
 #fi
 
 # GConf
-AC_PATH_PROG(GCONFTOOL, gconftool-2, no)
+if test "$gnome" != no ; then
+	AC_PATH_PROG(GCONFTOOL, gconftool-2, no)
+else
+	GCONFTOOL=no
+fi
 
 dnl *********************************************************************
 dnl ** XFT **************************************************************
@@ -329,9 +335,13 @@
 dnl ** PYTHON ***********************************************************
 dnl *********************************************************************
 
-if test "$python" = yes; then
-        AC_PATH_PROG(pythonpath, python2)
-        if test "_$pythonpath" = _ ; then
+if test "x$python" != xno ; then
+	if test "$python" = yes ; then
+		# default
+		python="python2"
+	fi
+        AC_PATH_PROG(pythonpath, $python)
+       	if test "_$pythonpath" = _ ; then
                 AC_PATH_PROG(pythonpath, python)
         fi
         if test "_$pythonpath" = _ ; then
@@ -358,7 +368,7 @@
                                 AC_MSG_RESULT([Can't find Python.h])
                         fi
                 else
-                        echo "Python too old. Only 2.2 or above is supported."
+                        echo "Python is too old or too new. Only 2.2-2.7 are supported."
                         python=no
                 fi
         fi
@@ -691,7 +701,7 @@
 AM_CONDITIONAL(DO_TEXT, test "x$textfe" = "xyes")
 AM_CONDITIONAL(DO_GTK, test "x$gtkfe" = "xyes")
 AM_CONDITIONAL(DO_PERL, test "x$perl" = "xyes")
-AM_CONDITIONAL(DO_PYTHON, test "x$python" = "xyes")
+AM_CONDITIONAL(DO_PYTHON, test "x$python" != "xno")
 AM_CONDITIONAL(DO_TCL, test "x$tcl" = "xyes")
 AM_CONDITIONAL(DO_PLUGIN, test "x$plugin" = "xyes")
 AM_CONDITIONAL(DO_CHECKSUM, test "x$checksum" = "xyes")
@@ -964,8 +974,6 @@
 plugins/doat/Makefile
 plugins/fishlim/Makefile
 plugins/sysinfo/Makefile
-intl/Makefile
-po/Makefile.in
 man/Makefile
 ])
 dnl plugins/sasl/Makefile
--- hexchat-2.9.4/Makefile.am
+++ hexchat-2.9.4/Makefile.am
@@ -1,13 +1,15 @@
 ## Process this file with automake to produce Makefile.in
 
-SUBDIRS = po intl src plugins man
+SUBDIRS = po src plugins man
 
 EXTRA_DIST = autogen.sh
 
+if DO_GTK
 icon_DATA = src/pixmaps/hexchat.png
-icondir = $(datadir)/pixmaps
+icondir = $(datadir)/icons/hicolor/48x48/apps
 
 util_DATA = share/misc/hexchat.desktop
 utildir = $(datadir)/applications
+endif
 
 ACLOCAL_AMFLAGS = -I m4
--- hexchat-2.9.4/src/common/Makefile.am
+++ hexchat-2.9.4/src/common/Makefile.am
@@ -52,6 +52,10 @@
 endif
 SUBDIRS = $(dbusdir) .
 
+if DO_PLUGIN
+include_HEADERS = hexchat-plugin.h
+endif
+
 libhexchatcommon_a_SOURCES = cfgfiles.c chanopt.c ctcp.c dcc.c hexchat.c \
 	history.c ignore.c inbound.c modes.c msproxy.c network.c notify.c \
 	outbound.c plugin.c plugin-timer.c proto-irc.c server.c servlist.c \
